docs/two-phase-commit-transactions.md

```diff
@@ -183,9 +183,49 @@ Similar to `prepare()`, you can call `validate()` in the coordinator/participant
 In other cases, `validate()` does nothing.
 
-### Suspend and resume the transaction
+### Load balancing in Two-phase Commit Transactions
+
+Usually, services using Two-phase Commit Transactions achieve a transaction while exchanging multiple requests/responses as follows:
+
+<p align="center">
+<img src="images/two_phase_commit_sequence_diagram.png" width="400" />
+</p>
+
+Also, a service typically has multiple hosts/servers for scalability and availability, and uses server-side (proxy) or client-side load balancing to distribute the requests to the hosts/servers.
+Since a transaction processing in Two-phase Commit Transactions is stateful, operations in a transaction must be executed in the same hosts/servers.
+In other words, the load balancing must distribute the requests in a transaction to the same host/server.
+
+<p align="center">
+<img src="images/two_phase_commit_load_balancing.png" width="500" />
+</p>
+
+As solutions to this issue depend on the protocol between the services, we introduce some solutions for gRPC and HTTP/1.1 here.
+
+#### gPRC
+
+Please see [this document](https://grpc.io/blog/grpc-load-balancing/) for the details of gRPC Load Balancing.
+
+When you use a client-side load balancer, you can use the same gRPC connection to send requests in a transaction, which guarantees that the requests go to the same hosts/servers.
+
+When you use a server-side (proxy) load balancer, solutions are different between when using L3/L4 (transport level) and L7 (application level) load balancer.
+When using an L3/L4 load balancer, you can use the same gRPC connection to send requests in a transaction, similar to when you use a client-side load balancer.
+Requests in the same gRPC connection always go to the same host/server in L3/L4 load balancing.
+When using an L7 load balancer, since requests in the same gRPC connection do not necessarily go to the same host/server, you need to use cookies or similar for routing requests to correct host/server.
+For example, when you use [Envoy](https://www.envoyproxy.io/), you can use session affinity (sticky session) for gRPC.
+Or you can also use [Bidirectional streaming RPC in gRPC](https://grpc.io/docs/what-is-grpc/core-concepts/#bidirectional-streaming-rpc) since the L7 load balancer distributes requests in the same stream to the same host/server.
+
+#### HTTP/1.1
+
+Typically, you use a server-side (proxy) load balancer with HTTP/1.1.
+When using an L3/L4 load balancer, you can use the same HTTP connection to send requests in a transaction, which guarantees the requests go to the same host/server.
+When using an L7 load balancer, since requests in the same HTTP connection do not necessarily go to the same host/server, you need to use cookies or similar for routing requests to correct host/server.
+You can use session affinity (sticky session) in that case.
+
+#### Suspend and resume a transaction
```

Comment:
This is a question. I was just curious.
I could not read a clear reason why I need to suspend the Tx.
Regarding the suspend/resume operations, if I don't suspend Tx as follows, what problems will occur?

```java
public class ServiceBImpl implements ServiceB {

  private TwoPhaseCommitTransactionManager manager = ...;

  ...

  @Override
  public void endpoint1(String txId) throws Exception {
    // First, you need to join the transaction
    TwoPhaseCommitTransaction tx = manager.join(txId);
    ...
  }

  @Override
  public void endpoint2(String txId) throws Exception {
    // You can resume the transaction suspended in endpoint1()
    TwoPhaseCommitTransaction tx = manager.resume(txId);
    ...
  }
...
```

In the above example, will `manager.resume(txId)` in the `endpoint2` return any error? (Since the Tx was not suspended in the previous API `endpoint1`.)

Reply:
Thank you for taking a look at this!

> In the above example, will manager.resume(txId) in the endpoint2 return any error? (Since the Tx was not suspended in the previous API endpoint1.)

Yes, `manager.resume(txId)` in the `endpoint2` throws an exception in that case.

Reply:
Thank you for your answer!
I understand.